### PR TITLE
Confluence connector wave 1

### PR DESCRIPTION
**Pull request by flvndvd**
This pull request lays the foundational structure for the Confluence connector, aligning with the specifications outlined [here](https://www.notion.so/DesignDoc-Confluence-integration-db8c55585c85464bbd8a89bbb42eaa8a?d=46bb02d6bdcb4f3e81f1aaa905ffba3b#f8ac8cd89dd9457e98aeb0fc569c78ea). Key highlights of this PR include:

- Creation of a `confluence_configurations` table to store `cloudId`, essential for constructing the Atlassian API endpoint.
- Establishment of a `confluence_spaces` table for managing selected **global** spaces.
- In accordance with the specifications, and due to limited granularity provided by the OAuth flow, this PR focuses on the retrieval, listing, and selection of global (non-private) spaces.
- Implementation of necessary groundwork for future plans, ensuring that even though not ready for production, the structure is set in place.
- Access to the Confluence connector is restricted to Dust employees and development environment.
- Development of a basic Confluence API wrapper, compensating for the absence of an official SDK from Atlassian.

Prior to the final release of this PR, the following tasks remain:
- [x] Establish a production application within the Atlassian Developer Console.
- [x] Add the secret `NANGO_CONFLUENCE_CONNECTOR_ID` in k8s secrets.

Subsequent PRs will follow to add the ingestion of pages within the selected spaces.

confluence-connector-oauth

### CODE REVIEW

**Comment by spolu** [comment]
>  Securely store the application secret within our Kubernetes cluster.

Which secret is that? Isn't that a nango thing? Was the `NANGO_CONFLUENCE_CONNECTOR_ID` added to front/connectors secrets?

File: connectors/src/admin/db.ts [hunk 1]
```diff
@@ -42,6 +46,8 @@ import logger from "@connectors/logger/logger";
 
 async function main(): Promise<void> {
   await Connector.sync({ alter: true });
+  await ConfluenceConfigurations.sync({ alter: true });
```
Comment (spolu) [inline connectors/src/admin/db.ts:49]: Table names are plural but models are singular. Not respected everywhere but that's what we should aim for :pray:

File: connectors/src/connectors/confluence/index.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      sourceUrl: `${confluenceConfig.url}/wiki${s._links.webui}`,
+      expandable: false,
+      permission: isSynced ? "read" : "none",
+      dustDocumentId: null,
+      lastUpdatedAt: null,
+    };
+  });
+
+  // List synced spaces.
+  if (filterPermission === "read") {
```
Comment (spolu) [inline connectors/src/connectors/confluence/index.ts:172]: Can we not do a call to Confluence if we're in filterPermission = read to speed up a notch the selection of resources in assistants?

**Comment by flvndvd** [comment]
> > Securely store the application secret within our Kubernetes cluster.
> 
> Which secret is that? Isn't that a nango thing? Was the `NANGO_CONFLUENCE_CONNECTOR_ID` added to front/connectors secrets?

Yes this todo is to add the `NANGO_CONFLUENCE_CONNECTOR_ID` in the `front-secrets`.

**Review by spolu — COMMENTED** [review]
First small batch of comments (leaving home for work) will continue when at work

**Comment by spolu** [comment]
> > > Securely store the application secret within our Kubernetes cluster.
> > 
> > 
> > Which secret is that? Isn't that a nango thing? Was the `NANGO_CONFLUENCE_CONNECTOR_ID` added to front/connectors secrets?
> 
> Yes this todo is to add the `NANGO_CONFLUENCE_CONNECTOR_ID` in the `front-secrets` before deploying.

Let's add that to the deploy procedure of the PR

File: connectors/src/admin/db.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/src/admin/db.ts:49]: Ok was not sure about the standards, but will update.

File: connectors/src/connectors/confluence/index.ts (hunk 2 above)
Comment (flvndvd) [reply connectors/src/connectors/confluence/index.ts:158]: Fixed.

File: connectors/src/connectors/confluence/index.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+      return connector;
+    });
+
+    // TODO(2024-01-10 flav) Uncomment in next PR.
+    // await launchConfluenceFullSyncWorkflow(connector.id, null);
+
+    return new Ok(connector.id.toString());
+  } catch (e) {
+    logger.error({ error: e }, "Error creating confluence connector.");
```
Comment (spolu) [inline connectors/src/connectors/confluence/index.ts:80]: Would be nice to add more info to the log Maybe creating a logger child with workspaceId (which is in datasourceconfig I thnk) / connectionId

File: front/lib/connector_providers.ts [hunk 4]
```diff
@@ -20,8 +20,23 @@ export const CONNECTOR_CONFIGURATIONS: Record<
...
     isNested: boolean;
+    dustWorkspaceOnly?: boolean;
   }
 > = {
+  confluence: {
+    name: "Confluence",
+    connectorProvider: "confluence",
+    isBuilt: true,
+    hide: false,
+    logoPath: "/static/intercom_32x32.png",
```
Comment (spolu) [inline front/lib/connector_providers.ts:31]: cc @Duncid

**Review by spolu — APPROVED** [review]
LGTM :100:

**Pull request merged**